packages/actor-rdf-resolve-quad-pattern-string-source/lib/ActorRdfResolveQuadPatternStringSource.ts

```diff
@@ -37,9 +41,26 @@ export class ActorRdfResolveQuadPatternStringSource extends ActorRdfResolveQuadP
     return true;
   }
 
-  public async run(action: IActionRdfResolveQuadPattern): Promise<IActorRdfResolveQuadPatternOutput> {
-    const source = <IDataSourceSerialized>getContextSource(action.context);
+  public run(action: IActionRdfResolveQuadPattern): Promise<IActorRdfResolveQuadPatternOutput> {
+    const source = <IDataSourceSerialized>getContextSource(action.context)!;
+    const rdfSourcePromise: Promise<RDF.Source> = this.cache?.get(source) ?? this.getRdfSource(action.context, source);
+    if (this.cache && !this.cache.has(source)) {
```

Comment:
This `this.cache.has()` call shouldn't be needed here.

Something like this should work:
```
let rdfSourcePromise: Promise<RDF.Source> = this.cache?.get(source);
if (!rdfSourcePromise) {
  // call getRdfSource and set in cache.
}
```

Reply:
This should be a bit better now after the change.